src/mixins/filing-template-mixin.ts

```diff
@@ -116,6 +117,175 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
 
+  /**
+   * Builds an amalgamation filing from store data. Used when saving a filing.
+   * @returns the filing body to save
+   */
+  buildAmalgamationFiling (): AmalgamationFilingIF {
+    function fixNullAddressType (orgPeople: OrgPersonIF[]): OrgPersonIF[] {
+      return orgPeople.map(p => {
+        if (p.deliveryAddress?.addressType === null) delete p.deliveryAddress.addressType
+        if (p.mailingAddress?.addressType === null) delete p.mailingAddress.addressType
+        return p
+      })
+    }
+
+    // Build the main filing.
+    const filing: AmalgamationFilingIF = {
+      header: {
+        name: FilingTypes.AMALGAMATION,
+        certifiedBy: this.getCertifyState.certifiedBy,
+        date: this.getCurrentDate,
+        filingId: this.getFilingId,
+        folioNumber: this.getFolioNumber,
+        isFutureEffective: this.getEffectiveDateTime.isFutureEffective
+      },
+      business: {
+        legalType: this.getEntityType,
+        identifier: this.getTempId
+      },
+      amalgamation: {
+        amalgamationType: this.getAmalgamationType,
+        nameRequest: {
+          legalType: this.getEntityType
+        },
+        nameTranslations: this.getNameTranslations,
+        offices: this.getDefineCompanyStep.officeAddresses,
+        contactPoint: {
+          email: this.getBusinessContact.email,
+          phone: this.getBusinessContact.phone,
+          // don't save extension if it's empty
+          ...this.getBusinessContact.extension
+            ? { extension: +this.getBusinessContact.extension }
+            : {}
+        },
+        parties: fixNullAddressType(this.getAddPeopleAndRoleStep.orgPeople)
+      }
+    }
+
+    filing.amalgamation.shareStructure = {
+      shareClasses: this.getCreateShareStructureStep.shareClasses
+    }
+    filing.amalgamation.incorporationAgreement = {
+      agreementType: this.getIncorporationAgreementStep.agreementType
+    }
+
+    const courtOrder = this.getCourtOrderStep.courtOrder
+    if (courtOrder && (courtOrder.hasPlanOfArrangement || courtOrder.fileNumber)) {
+      filing.amalgamation.courtOrder = {
+        fileNumber: courtOrder.fileNumber,
+        effectOfOrder: courtOrder.hasPlanOfArrangement ? EffectOfOrders.PLAN_OF_ARRANGEMENT : '',
+        hasPlanOfArrangement: courtOrder.hasPlanOfArrangement
+      }
+    }
+
+    // If this is a named IA then add Name Request Number and Approved Name.
+    if (this.getNameRequestNumber) {
+      filing.amalgamation.nameRequest.nrNumber = this.getNameRequestNumber
+      filing.amalgamation.nameRequest.legalName = this.getNameRequestApprovedName
+    }
+
+    // If this is a future effective filing then save the effective date.
+    if (this.getEffectiveDateTime.isFutureEffective) {
+      filing.header.effectiveDate = this.dateToApi(this.getEffectiveDateTime.effectiveDate)
+    }
+
+    if (this.isRoleStaff) {
+      // Add staff payment data.
+      this.buildStaffPayment(filing)
+    }
+    return filing
+  }
```

Comment:
There's probably some stuff missing above for new amalgamation-specific data.

Ditto below for parsing a draft.

This is TODO for whomever works on the new components/data.